engine/common/rpc/convert/execution_data.go

```diff
@@ -316,6 +319,42 @@ func messageToTrustedTransaction(
 	return *t, nil
 }
 
+func MessageToRegisterID(m *entities.RegisterID) (flow.RegisterID, error) {
+	if m == nil {
+		return flow.RegisterID{}, ErrEmptyMessage
+	}
+	return flow.RegisterID{
+		Owner: m.GetOwner(),
+		Key:   m.GetKey(),
+	}, nil
+}
+
+// MessagesToRegisterIDs converts a protobuf message to RegisterIDs
+func MessagesToRegisterIDs(m []*entities.RegisterID) (flow.RegisterIDs, error) {
+	if m == nil {
+		return nil, ErrEmptyMessage
+	}
+	if len(m) > MaxRegisterIDsPerRequest {
```

Comment:
this check should go in the backend logic. convert is only responsible for translating between formats